### PR TITLE
Event dao

### PR DESCRIPTION
**Pull request by volodymyrvynnyk**

### CODE REVIEW

File: src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@Repository
+public class EventDaoImpl implements EventDao {
+    private static Logger logger = LoggerFactory.getLogger(UserDaoImpl.class);
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java]: make it final

File: src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    public EventDaoImpl(JdbcTemplate jdbcTemplate, EventRowMapper eventRowMapper) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.eventRowMapper = eventRowMapper;
+    }
+
+    @Override
+    public Event create(Event event) {
+        try {
+            event.setDate(ZonedDateTime.now(ZoneId.systemDefault()));
+            jdbcTemplate.update(Queries.SQL_CREATE_EVENT, event.getMessage(), event.getDate().toLocalDateTime(),
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java]: Donesn't it work with ZoneDateTime?

Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java]: id of the event is not retrieved after insert

File: src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+        static final String SQL_SELECT_EVENTS_BY_WAREHOUSE_ID_AND_TYPE = "SELECT* FROM events WHERE warehouse_id = ? AND type = ?";
+
+        static final String SQL_SELECT_EVENTS_BY_WAREHOUSE_ID_AND_DATE = "SELECT* FROM events WHERE warehouse_id = ? AND DATE(date) = ?";
+
+        static final String SQL_SELECT_EVENTS_BY_ACCOUNT_ID_AND_DATE = "SELECT* FROM events WHERE account_id = ? AND DATE(date) = ?";
+
+        static final String SQL_SELECT_EVENTS_BY_ACCOUNT_ID_AND_WITHIN_DATES = "SELECT* FROM events WHERE account_id = ? AND DATE(date) >= ? AND DATE(date) <= ?";
+
+        static final String SQL_SELECT_EVENTS_BY_WAREHOUSE_ID_AND_WITHIN_DATES = "SELECT* FROM events WHERE warehouse_id = ? AND DATE(date) >= ? AND DATE(date) <= ?";
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java]: SQL Syntax exception, There is no space between SELECT and *

File: src/main/java/com/ita/if103java/ims/mapper/jdbc/EventRowMapper.java [hunk 4]
```diff
@@ -16,12 +18,12 @@ public Event mapRow(ResultSet resultSet, int i) throws SQLException {
         Event event = new Event();
         event.setId(resultSet.getLong("id"));
         event.setMessage(resultSet.getString("message"));
-        event.setDate(resultSet.getObject("date", ZonedDateTime.class));
+        event.setDate(ZonedDateTime.of(resultSet.getObject("date", LocalDateTime.class), ZoneId.systemDefault()));
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/mapper/jdbc/EventRowMapper.java]: Does it not work with ZoneDateTime?

Comment (volodymyrvynnyk) [reply src/main/java/com/ita/if103java/ims/mapper/jdbc/EventRowMapper.java]: I had tried, but I couldn't insert ZonedDateTime into table

**Pull request merged**